module/events/finalization_actor.go

```diff
@@ -0,0 +1,70 @@
+package events
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/tracker"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+)
+
+// OnBlockFinalized is invoked when a new block is finalized. It is possible that
+// blocks will be skipped.
+type OnBlockFinalized func(block *model.Block) error
```

Comment:
To me, the naming here seems problematic:
* `OnBlockFinalized` is different yet not easily distinguishable from the [`OnFinalizedBlock` notification](https://github.com/onflow/flow-go/blob/a6f6fd591aa6d776b50ec47c6d83c79461a9eb20/consensus/hotstuff/consumer.go#L28-L33)
* I would guess that the difference is too small to notice for most people (I didn't realize that both methods were different until quite a bit through this PR). 

I could imagine two different approaches:
* We name it identically to the [FinalizationConsumer's `OnFinalizedBlock` notification](https://github.com/onflow/flow-go/blob/a6f6fd591aa6d776b50ec47c6d83c79461a9eb20/consensus/hotstuff/consumer.go#L28-L33). Then, we have the benefits of being able to search for the method
* If we want to highlight that they are different (this method is allowed to block and notifications can be dropped), I would make the methods' names obviously different.
   * For example, we could name this `ProcessLatestFinalizedBlock` (not sure if the "latest" provides enough of a hint that blocks could be skipped)

Additionally, I have a soft preference to use an interface here over anonymous methods.